src/box/tuple.c

```diff
@@ -143,15 +150,21 @@ tuple_extract_key(const struct tuple *tuple, const struct key_def *key_def,
 		return NULL;
 	}
 	char *key_buf = mp_encode_array(key, part_count);
-	for (uint32_t i = 0; i < part_count; ++i) {
+	for (uint32_t i = 0; i < part_count;) {
+		uint32_t len = key_def_seq_len(key_def, key_def->parts[i].fieldno);
 		const char *field =
 			tuple_field_raw(format, data, field_map,
```

Comment:
This patch optimize nothing, because tuple_field_raw() is called even when key_def is sequential.